Dockerfile

```diff
@@ -1,22 +1,51 @@
-FROM rust:1.80-alpine AS build
+FROM --platform=${BUILDPLATFORM} ghcr.io/cross-rs/aarch64-unknown-linux-musl:0.2.5 AS build-arm64
```

Comment:
Using the `rs-cross` [project](https://github.com/cross-rs/cross) container image I can cross compile the code inside the container image. This container image already have all the tools required by the toolchain/target to compile the code (e.g. aarch64 compiler). This allows us to build the project in 20 minutes instead of 3 hours.

Reply:
This sounds great!
I wonder though if it is possible to remove the duplication, we are building on top of both build-arm64 and build-am64, even if at the end we only copy the relevant binary to scratch. Both the buildchains may appear in provenance and so, when there's only need of 1.

Reply:
Take this [CI](https://github.com/jvanz/policy-server/actions/runs/11294413218) x86_64 provenance file. The build commands does not show reference of usage of the aarch container image. The aarch container image is mentioned in the "materials" field only. But I do not know if this is an issue or not. 

```console
jq ".predicate.materials" policy-server-attestation-amd64-provenance.json
[
  {
    "uri": "pkg:docker/docker/buildkit-syft-scanner@stable-1",
    "digest": {
      "sha256": "176e0869c38aeaede37e594fcf182c91d44391a932e1d71e99ec204873445a33"
    }
  },
  {
    "uri": "pkg:docker/alpine@latest?platform=linux%2Famd64",
    "digest": {
      "sha256": "beefdbd8a1da6d2915566fde36db9db0b524eb737fc57cd1367effd16dc0d06d"
    }
  },
  {
    "uri": "pkg:docker/ghcr.io/cross-rs/aarch64-unknown-linux-musl@0.2.5?platform=linux%2Famd64",
    "digest": {
      "sha256": "702154f52b2d8091671aa2c84d5582d849f949977228c735ff8462f93cc0e1e4"
    }
  },
  {
    "uri": "pkg:docker/ghcr.io/cross-rs/x86_64-unknown-linux-musl@0.2.5?platform=linux%2Famd64",
    "digest": {
      "sha256": "77db671d8356a64ae72a3e1415e63f547f26d374fbe3c4762c1cd36c7eac7b99"
    }
  }
]

```